server/src/main/java/org/apache/iotdb/db/query/executor/fill/ValueFill.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.query.executor.fill;
+
+import org.apache.iotdb.db.metadata.PartialPath;
+import org.apache.iotdb.db.query.context.QueryContext;
+import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.TimeValuePair;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.utils.TsPrimitiveType;
+
+import java.util.Set;
+
+public class ValueFill extends IFill implements Cloneable {
+
+  private String value;
+
+  private boolean booleanValue;
+  private int intValue;
+  private long longValue;
+  private float floatValue;
+  private double doubleValue;
+  private Binary textValue;
+
+  public ValueFill(String value, TSDataType dataType) {
+    this.value = value;
+    this.dataType = dataType;
+    parseValue();
+  }
+
+  @Override
+  public IFill copy() {
+    return (IFill) clone();
+  }
+
+  @Override
+  public Object clone() {
+    ValueFill valueFill = null;
+    try {
+      valueFill = (ValueFill) super.clone();
+    } catch (CloneNotSupportedException e) {
+    }
+    return valueFill;
+  }
+
+  @Override
+  public void configureFill(
+      PartialPath path,
+      TSDataType dataType,
+      long queryTime,
+      Set<String> deviceMeasurements,
+      QueryContext context) {}
+
+  @Override
+  public TimeValuePair getFillResult() {
+    switch (dataType) {
+      case BOOLEAN:
+        return new TimeValuePair(queryTime, new TsPrimitiveType.TsBoolean(booleanValue));
```

Comment:
could we cache TsPrimitiveType not the primary type?

Reply:
Tsprimitivetype can be cached